src/warplib/implementations/maths/exp.ts

```diff
@@ -35,18 +45,38 @@ function createExp(signed: boolean, unsafe: boolean) {
     forAllWidths((width) => {
       if (width === 256) {
         return [
-          `func _repeated_multiplication${width}{range_check_ptr, bitwise_ptr : BitwiseBuiltin*}(op : Uint256, count : Uint256) -> (res : Uint256):`,
+          `func _repeated_multiplication${width}{range_check_ptr, bitwise_ptr : BitwiseBuiltin*}(op : Uint256, count : felt) -> (res : Uint256):`,
+          `    if count == 0:`,
+          `        return (Uint256(1, 0))`,
+          `    end`,
+          `    let (x) = _repeated_multiplication${width}(op, count - 1)`,
```

Comment:
No. of steps can be significantly reduced if we use logarithmic complexity repeated multiplication instead of linear one. 
```cairo
let (y) = _repeated_multiplication${width}(op, count/2)
let (x) = y*y
if count is even : let res = x
else :  let res = warp_mul${suffix}${width}(op, x)

```

Reply:
This is definitely worth looking into post audit to see how much it does